DevelopingSpace/starchart
# Setup test database for running tests for database

Part of #238
This sets up a test database `starchart_test` in the container `test_database`, and it can be used to run tests without interfering with the other database container or any other database.

For convenience, I added a `.env.test` file, for devs to run `cp .env.test .env` to change the database connection string to the test database.

## Changes
* Add test database in in `docker-compose.yml`
* Add  `.env.test`
* Ignore the volume at `/mysql-data-test` for git, Prettier, ESlint, and Docker.

## Testing
1. Run `cp .env.test .env`
2. Run `docker compose up -d` to make containers like usual
3. Run `npm run steup` to connect, push schema, and seed the database
4. Run `npm run db:studio` to check that the database works
